flutter/lib/ui/geometry.dart

```diff
@@ -820,10 +951,24 @@ class RRect {
   bool get isEmpty => left >= right || top >= bottom;
 
   /// Whether this rounded rectangle has a side with no straight section.
-  bool get isStadium => width <= 2 * radiusX || height <= 2 * radiusY;
+  bool get isStadium {
+    return (
+      height <= blRadiusY + tlRadiusY ||
+      width <= tlRadiusX + trRadiusX ||
+      height <= trRadiusY + brRadiusY ||
+      width <= brRadiusX + blRadiusX
+    );
+  }
 
   /// Whether this rounded rectangle has no side with a straight section.
-  bool get isEllipse => width <= 2 * radiusX && height <= 2 * radiusY;
+  bool get isEllipse {
```

Comment:
this should also check that the x radii are equal and the y radii are equal (i think that's what is needed to make an ellipse, right?)